lib/model/model.go

```diff
@@ -1244,52 +1244,48 @@ func (m *Model) GetIgnores(folder string) ([]string, []string, error) {
 	m.fmut.RLock()
 	cfg, ok := m.folderCfgs[folder]
 	m.fmut.RUnlock()
-	if !ok {
-		cfg, ok = m.cfg.Folders()[folder]
-		if ok {
-			matcher := ignore.New(false)
-			path := filepath.Join(cfg.Path(), ".stignore")
-			if err := matcher.Load(path); err != nil {
-				return nil, nil, err
-			}
-			return matcher.Lines(), matcher.Patterns(), nil
+	if ok {
+		if !cfg.HasMarker() {
+			return nil, nil, fmt.Errorf("Folder %s stopped", folder)
 		}
-		return nil, nil, fmt.Errorf("Folder %s does not exist", folder)
-	}
 
-	if !cfg.HasMarker() {
-		return nil, nil, fmt.Errorf("Folder %s stopped", folder)
+		m.fmut.RLock()
+		ignores := m.folderIgnores[folder]
+		m.fmut.RUnlock()
+
+		return ignores.Lines(), ignores.Patterns(), nil
 	}
 
-	m.fmut.RLock()
-	lines := m.folderIgnores[folder].Lines()
-	patterns := m.folderIgnores[folder].Patterns()
-	m.fmut.RUnlock()
+	if cfg, ok := m.cfg.Folders()[folder]; ok {
+		matcher := ignore.New(false)
+		path := filepath.Join(cfg.Path(), ".stignore")
+		if err := matcher.Load(path); err != nil {
+			return nil, nil, err
+		}
+		return matcher.Lines(), matcher.Patterns(), nil
+	}
 
-	return lines, patterns, nil
+	return nil, nil, fmt.Errorf("Folder %s does not exist", folder)
 }
 
 func (m *Model) SetIgnores(folder string, content []string) error {
-	m.fmut.RLock()
-	cfg, ok := m.folderCfgs[folder]
-	m.fmut.RUnlock()
-	var paused bool
+	cfg, ok := m.cfg.Folders()[folder]
 	if !ok {
-		cfg, paused = m.cfg.Folders()[folder]
-		if !paused {
-			return fmt.Errorf("Folder %s does not exist", folder)
-		}
+		return fmt.Errorf("Folder %s does not exist", folder)
 	}
 
 	if err := ignore.WriteIgnores(filepath.Join(cfg.Path(), ".stignore"), content); err != nil {
 		l.Warnln("Saving .stignore:", err)
 		return err
 	}
 
-	if paused {
-		return nil
+	m.fmut.RLock()
+	_, ok = m.folderCfgs[folder]
```

Comment:
You can get the runner, which would have a scan method on it, instead of calling model.Scan() (which does that anyway).